### PR TITLE
copyInterfaceInputs bug fix

### PR DESCRIPTION
**Pull request by opotowsky**
## Description

`copyInterfaceInputs` was not properly handling the use case of a list of filenames. It worked only when a single input file was associated with a setting. 

The primary logic change in this PR is changing the update being a single file: 
```python
newSettings[label] = str(destFilePath)
```
To the option for the update being a list: 
```python
newSettings[label].append(str(destFilePath))
```

Additionally, the handling of wildcards has moved around and a check for that has been implemented by setting `globFilePaths` to `None` and only processing a glob if that variable exists.

I should have code coverage COVERED. Several added tests. 

This function should now be able to handle the updating of a single file (relative or absolute), a list of files (relative or absolute), and a file entry that has a wildcard processing into multiple files. Although, that makes it a bit [smelly](https://en.wikipedia.org/wiki/Code_smell) (Thanks for the reference @sombrereau!).

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [ ] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).
**(Note to PR merger: the first commit message is bad so please grab the first line of the PR description when squash-and-merging instead!)**

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by jakehader** [comment]
@opotowsky, this is a good fix! I went to review but noticed that it was still a draft. 

I know that this issue (https://github.com/terrapower/armi/issues/772) was opened previously, but I'm wondering if it's still an issue that users (i.e., @keckler) are still seeing. I tend to think that we should have `copyInterfaceInputs` be based on already defined case settings, but maybe that is overly restrictive as in some cases we need to bring more than a single file along (thinking custom fuel management or custom control rod management scripts that have other dependencies). 

I like one of the unit tests that @john-science is able to add that demonstrates a custom plugin, so maybe we can create a lightweight plugin with an interface that utilizes this `copyInterfaceInputs` for improved test coverage from end to end. Don't want to detail your progress, but it might be nice to close the loop on that other issue at the same time as this? See - #796

**Comment by opotowsky** [comment]
@jakehader I actually am not addressing that issue in this PR, although it is certainly on my todo list (it became very time intensive). @keckler has a workaround but he convinced me of the need to support non-interface-related file clone support, so I self-assigned the issue since I broke downstream use cases. 

For this PR, I ran into a use case where an interface has a setting that was a list of filenames, which this function implied support for, but did not actually support. 

I thought it was an easy fix but I didn't understand the reasons for the complexity in this function, so it was not in fact simple. It is draft until I get the tests fully figured out, as well as fix some code-quality issues. Thank you for the suggestion, I think it will help with progress and not derail it!

**Comment by opotowsky** [comment]
Marking as draft--again

File: armi/cases/tests/test_cases.py [hunk 1]
```diff
@@ -272,20 +277,159 @@ def test_writeInput(self):
             self.assertTrue(os.path.exists(cs["shuffleLogic"]))
 
 
+class MultiFilesInterfaces(interfaces.Interface):
+    """
+    A mostly meaningless little test interface, just to prove that TODO
```
Comment (john-science) [inline armi/cases/tests/test_cases.py:282]: Damn. Sorry, I put a TODO in your code. My bad.

```
A little test interface that adds a setting that we need to test copyInterfaceInputs
with multiple files.
```

Comment (john-science) [reply armi/cases/tests/test_cases.py:282]: My fault!

Comment (opotowsky) [reply armi/cases/tests/test_cases.py:282]: Good catch! on it

**Review by john-science — APPROVED** [review]
Nice work!

Man, that function is just _too_ complicated.

**Comment by opotowsky** [comment]
> Nice work!
> 
> Man, that function is just _too_ complicated.

My work here is not done. Until tomorrow!

**Pull request merged**